src/parsers/PnmlParser.js

```diff
@@ -1,92 +1,126 @@
             }
-        }
+        });
+
+        // Make sure elements are created in the right order
+        this.sortElements();
 
         return {
-            elements,
+            elements: this.elements,
             title,
         };
     }
 
-    createElement (node) {
-        const element = this.metaFactory.createElement('pt:' + node.nodeName);
-        element.id = 'import_' + node.attributes.id.value;
+    createElement (net, elementName) {
+        const elementData = net[elementName];
+        const element = this.metaFactory.createElement('pt:' + elementName);
+        this.elements.push(element);
+
+        element.id = 'import_' + elementData._attributes.id;
         element.parentId = '__implicitroot';
 
-        if (node.nodeName === 'arc') {
-            element.sourceId = 'import_' + node.attributes.source.value;
-            element.targetId = 'import_' + node.attributes.target.value;
+        if (elementData.graphics && elementData.graphics.position) {
+            element.x = elementData.graphics.position._attributes.x || 0;
+            element.y = elementData.graphics.position._attributes.y || 0;
         }
 
-        return element;
-    }
-
-    setPosition (node, element) {
-        if (!element) {
-            return;
+        if (elementData.graphics && elementData.graphics.dimension) {
+            this.setDimension(element, elementData.graphics.dimension);
         }
 
-        element.x = parseInt(node.attributes.x.value);
-        element.y = parseInt(node.attributes.y.value);
+        switch (elementName) {
+            case 'place':
+                if (elementData.initialMarking) {
+                    // PNML standard
+                    this.createLabel(
+                        element,
+                        elementData.initialMarking,
+                        'pt:marking'
+                    );
+                } else if (elementData.toolspecific
+                    && elementData.toolspecific.inscription) {
+                    // Renew specific
+                    this.createLabel(
+                        element,
+                        elementData.toolspecific.inscription,
+                        'pt:marking'
+                    );
+                }
+                if (elementData.name) {
+                    this.createLabel(element, elementData.name, 'pt:name');
+                }
+                break;
+            case 'transition':
+                break;
+            case 'arc':
+                element.sourceId = 'import_' + elementData._attributes.source;
+                element.targetId = 'import_' + elementData._attributes.target;
+                break;
+        }
     }
 
-    setDimension (node, element) {
-        if (!element) {
-            return;
-        }
+    createLabel (element, labelData, labelType) {
+        const label = this.metaFactory.createElement(labelType);
+        label.width = 150; // TODO get default dimensions from somewhere
```

Comment:
This will be an additional issue, right? So we need to create an issue for this.

Reply:
I created #14 referencing [renew-lib#123](https://github.com/renew-js/renew-lib/issues/123) for this.